File: Luogu/Course/JCA/JC2/P_1042_NOIP_2003_普及组_乒乓球.py
run = True
s = ""
while run:
    line = input().strip()
    if not line:
        continue
    if "E" in line: # E 不一定是最後一個字元
        run = False
        line = line[:line.index("E")]
    s += line

for target in [11, 21]:
    x = y = 0
    for ch in s:
        if ch == "W":
            x += 1
        elif ch == "L":
            y += 1
        if max(x, y) >= target and abs(x - y) >= 2:
            print(f"{x}:{y}")
            x = y = 0
    print(f"{x}:{y}")
    print()

File: Luogu/Course/JCA/JC2/P_1042_NOIP_2003_普及组_乒乓球.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    char ch;
    string s, line;
    while (cin >> ch) {
        if (ch == 'E') {
            break;
        }
        s += ch;
    }
    for (int target : {11, 21}) {
        int x = 0, y = 0;
        for (char ch : s) {
            if (ch == 'W') x++;
            else y++;
            if (max(x, y) >= target && abs(x - y) >= 2) {
                cout << x << ":" << y << endl;
                x = y = 0;
            }
        }
        cout << x << ":" << y << endl;
        cout << endl;
    }
    return 0;
}